client/src/components/Form/FormCardTool.vue

```diff
@@ -0,0 +1,280 @@
+    components: {
+        FormMessage,
+        ToolFooter,
+        ToolHelp,
+    },
+    props: {
+        id: {
+            type: String,
+            required: true,
+        },
+        user: {
+            type: Object,
+            required: true,
+        },
+        version: {
+            type: String,
+            required: false,
+        },
+        title: {
+            type: String,
+            required: true,
+        },
+        description: {
+            type: String,
+            required: false,
+        },
+        sustainVersion: {
+            type: Boolean,
+            default: false,
+        },
+        options: {
+            type: Object,
+            required: true,
+        },
+        messageText: {
+            type: String,
+            required: true,
+        },
+        messageVariant: {
+            type: String,
+            default: "info",
+        },
+    },
+    data() {
+        return {
+            webhookDetails: [],
+            errorText: null,
+        };
+    },
+    watch: {
+        id() {
+            this.errorText = null;
+        },
+    },
+    computed: {
+        versions() {
+            return this.options.versions;
+        },
+        showAddFavorite() {
+            return !!this.user.email && !this.isFavorite;
+        },
+        showVersions() {
+            return !this.sustainVersion && this.versions && this.versions.length > 1;
+        },
+        availableVersions() {
+            const versions = this.versions.slice();
+            const index = versions.indexOf(this.version);
+            if (index > -1) {
+                versions.splice(index, 1);
+            }
+            return versions.reverse();
+        },
+        showDownload() {
+            return this.user && this.user.is_admin;
+        },
+        showLink() {
+            return this.options.sharable_url;
+        },
+        isFavorite() {
+            return this.getFavorites().tools.includes(this.id);
+        },
+    },
+    created() {
+        // add tool menu webhooks
+        Webhooks.load({
+            type: "tool-menu",
+            callback: function (webhooks) {
+                webhooks.each((model) => {
+                    const webhook = model.toJSON();
+                    if (webhook.activate && webhook.config.function) {
+                        this.webhookDetails.push({
+                            icon: `fa ${webhook.config.icon}`,
+                            title: webhook.config.title,
+                            onclick: function () {
+                                const func = new Function("options", webhook.config.function);
+                                func(this.options);
+                            },
+                        });
+                    }
+                });
+            },
+        });
+    },
+    methods: {
+        onAddFavorite() {
+            axios.put(`${getAppRoot()}api/users/${this.user.id}/favorites/tools`, { object_id: this.id }).then(
```

Comment:
Externalizing data calls would make for easier unit testing.

Reply:
Done.